components/script/dom/xmlhttprequest.rs

```diff
@@ -706,7 +706,10 @@ impl XMLHttpRequestMethods for XMLHttpRequest {
         // Step 2
         let override_mime = try!(mime.parse::<Mime>().map_err(|_| Error::Syntax));
         // Step 3
-        *self.override_mime_type.borrow_mut() = Some(override_mime.clone());
+        let type_str_no_params = mime.split(';').next().unwrap_or(&mime);
+        let lower_type_str = type_str_no_params.to_ascii_lowercase();
+        let mime_no_params = lower_type_str.parse::<Mime>().unwrap();
```

Comment:
I don't think we actually need to lowercase the string, since the Mime parsing is case-insensitive and will normalize everything to lowercase after parsing. What we do have to do instead is to create a new Mime struct and then set its param value to an empty vec, and then set `self.override_mime_type` to this new struct.